### PR TITLE
Add camera-lookat option

### PR DESCRIPTION
**Pull request by jafyvilla**
* The option takes a position vector in world space.
Using this value together with the camera-position,
the camera is transformed to point to the specified position.
* Also, set the initial camera-position as default.

### CODE REVIEW

File: livre/Eq/Settings/CameraSettings.cpp [hunk 1]
```diff
@@ -171,6 +174,11 @@ void CameraSettings::setCameraRotation( const Vector3f& rotation )
     setDirty( DIRTY_ALL );
 }
 
+void CameraSettings::setCameraLookAt( const Vector3f& lookAt )
+{
+    setModelviewMatrix( maths::computeModelViewMatrix( cameraPosition_, lookAt ));
```
Comment (tribal-tec) [inline livre/Eq/Settings/CameraSettings.cpp]: uppercase V

File: livre/Eq/Settings/CameraSettings.h [hunk 2]
```diff
@@ -49,6 +49,8 @@ class CameraSettings : public co::Serializable
      */
     void reset();
 
+    void setInitialCameraPosition( const Vector3f& position );
```
Comment (tribal-tec) [inline livre/Eq/Settings/CameraSettings.h]: doxygen; mention that 'Space' key goes back to this one

File: livre/Lib/Configuration/ApplicationParameters.cpp [hunk 3]
```diff
@@ -42,10 +42,12 @@ const std::string ANIMATION_PARAM = "animation";
 const std::string FRAMES_PARAM = "frames";
 const std::string NUMFRAMES_PARAM = "num-frames";
 const std::string CAMERAPOS_PARAM = "camera-position";
+const std::string CAMERALOOKAT_PARAM = "camera-look";
```
Comment (tribal-tec) [inline livre/Lib/Configuration/ApplicationParameters.cpp]: camera-lookat

**Comment by tribal-tec** [comment]
+1

File: livre/Eq/Settings/CameraSettings.cpp [hunk 4]
```diff
@@ -90,40 +97,32 @@ void CameraSettings::spinCamera( const float x, const float y )
...
-    modelRotation_.pre_rotate_y( y );
-    modelRotation_.pre_rotate_z( z );
+    Matrix4f matInverse;
+    cameraRotation_.inverse( matInverse );
+    Vector4f shift = matInverse * Vector4f( x, y, z, 1 );
+    modelRotation_.pre_rotate_x( shift.x( ));
+    modelRotation_.pre_rotate_y( shift.y( ));
+    modelRotation_.pre_rotate_z( shift.z( ));
 
     cameraSpin_ = sqrt( x*x + y*y + z*z );
```
Comment (bilgili) [inline livre/Eq/Settings/CameraSettings.cpp]: std::sqrt

**Comment by tribal-tec** [comment]
-1 :) Demo showed is does not work

File: livre/core/Maths/Maths.cpp [hunk 5]
```diff
@@ -61,26 +60,52 @@ void getRotationAndEyePositionFromModelView( const Matrix4f& modelviewMatrix,
...
-{
-    Matrix4f rotationTranspose = Matrix4f::IDENTITY;
-    rotationTranspose.set_sub_matrix( rotationMatrix, 0, 0 );
-    rotationTranspose = transpose( rotationTranspose );
-    Matrix4f modelviewMatrix = Matrix4f::IDENTITY;
-    modelviewMatrix.set_translation( -eye );
-    return rotationTranspose * modelviewMatrix;
-}
-
 Matrix4f computeModelViewMatrix( const Matrix4f& rotationMatrix, const Vector3f& eye )
```
Comment (bilgili) [inline livre/core/Maths/Maths.cpp]: rotationMatrix here is Matrix4f and it looks a bit fishy ( who did that I dont know ;) ). It should better be Matrix3f.

**Comment by tribal-tec** [comment]
+1

**Pull request merged**